git/git.go

```diff
@@ -459,7 +459,7 @@ func DefaultRemote() (string, error) {
 }
 
 func UpdateIndexFromStdin() *subprocess.Cmd {
-	return gitNoLFS("update-index", "-q", "--refresh", "--stdin")
+	return git("update-index", "-q", "--refresh", "--stdin")
```

Comment:
OK, here is my understanding:

We hit this code path *only* if we manually resolve the Git LFS pointers. Manually here means: we don't use the Git clean/smudge mechanism.  This happens in the `git lfs clone ...` and `git lfs pull` calls. In these calls the following happens:

1. Git checks out all files without running the Git LFS filters. We have the Git LFS pointer files checked out as a result.
2. Git LFS iterates over all pointer files and replaces them with the actual content
3. Git LFS asks Git to re-index the changed content (this happens with the `update-index` call above). During the re-index Git needs to `clean` the real content to calculate the pointer file and compare that against the version stored in the repository. This way `git status` can tell us if the file was changed or not.

--> I think it makes sense to run Git with the clean filter here.